### PR TITLE
Update SFIA engineering roles references

### PR DESCRIPTION
**Pull request by Joseaie**
Summary:
The role titles for `Software Engineer 1` and `Software Engineer 2` have been changed to `Associate Software Engineer` and `Software Engineer`, however, there were still references to their past selves in the SFIA documentation. So, I just looked for them in the codebase and changed them.

### CODE REVIEW

**Review by cob16 — APPROVED** [review]

**Comment by reub-the-cube** [comment]
Great stuff! Closes #678

**Review by reub-the-cube — APPROVED** [review]

**Comment by Joseaie** [comment]
Awesome; thanks for approving! @cob16  @reub-the-cube  
I can't merge it in though, as I don't have write access 👀

**Pull request merged**
